### PR TITLE
Gen.go update

### PR DESCRIPTION
**Pull request by mcardosos**
This works with the cowboy's latest changes.
PTAL @marstr @jhendrixMSFT

### CODE REVIEW

File: Gododir/gen.go [hunk 1]
```diff
@@ -404,14 +414,16 @@ func generate(service *service) {
 	fmt.Printf("Generating %s...\n\n", service.Fullname)
 	delete(service)
 
-	autorest := exec.Command(fmt.Sprintf("%s/autorest/src/core/AutoRest/bin/Debug/net451/win7-x64/autorest", autorestDir),
+	autorest := exec.Command("gulp",
```
Comment (marstr) [inline Gododir/gen.go]: Given that this is pretty new stuff, and some of our computers (okay, well mine) haven't had gulp setup yet. We may want to add in some guard rails like this:
[exec.LookPath(string)](https://godoc.org/os/exec#LookPath)

File: Gododir/gen.go [hunk 2]
```diff
@@ -404,14 +414,16 @@ func generate(service *service) {
...
+		"autorest",
 		"-Input", fmt.Sprintf("%s/azure-rest-api-specs/%s.json", swaggersDir, service.Input),
 		"-CodeGenerator", "Go",
 		"-Header", "MICROSOFT_APACHE",
 		"-Namespace", service.Name,
 		"-OutputDirectory", service.Output,
 		"-Modeler", "Swagger",
 		"-pv", sdkVersion)
+	autorest.Dir = fmt.Sprintf("%s/autorest", autorestDir)
```
Comment (marstr) [inline Gododir/gen.go]: We're all cross-platform like now! How about we update it to:

``` Go
autorest.Dir = filepath.Join(autores.Dir, "autorest")
```

[filepath.Join(string...)](https://golang.org/pkg/path/filepath/#Join)

File: Gododir/gen.go [hunk 3]
```diff
@@ -404,15 +415,22 @@ func generate(service *service) {
...
 
-	autorest := exec.Command(fmt.Sprintf("%s/autorest/src/core/AutoRest/bin/Debug/net451/win7-x64/autorest", autorestDir),
+	_, err := exec.LookPath("gulp")
+	if err != nil {
+		panic("You need gulp (and some other tools) to run AutoRest. See https://github.com/Azure/autorest/pull/1827")
+	}
+
+	autorest := exec.Command("gulp",
+		"autorest",
 		"-Input", fmt.Sprintf("%s/azure-rest-api-specs/%s.json", swaggersDir, service.Input),
```
Comment (jhendrixMSFT) [inline Gododir/gen.go]: Another candidate for filepath.Join.

**Comment by mcardosos** [comment]
Wait, it is not working as I expected yet :I

File: Gododir/gen.go (hunk 3 above)
Comment (mcardosos) [inline Gododir/gen.go]: Fixed (and used filepath.Join in some other places too)

File: Gododir/gen.go (hunk 1 above)
Comment (mcardosos) [inline Gododir/gen.go]: Fixed

**Comment by mcardosos** [comment]
It works!

**Pull request merged**